src/App.tsx

```diff
@@ -1,61 +1,139 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUserById(id: number, users: User[]) {
+  return users.find(user => id === user.id) || null;
+}
+
+const todoList: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId, usersFromServer),
+}));
+
+function getUserByName(name: string, users: User[]) {
+  return users.find(user => name === user.name) || null;
+}
+
+function getLargestTodoId(todos: Todo[]) {
+  return Math.max(...todos.map(todo => todo.id));
+}
+
+export const App: React.FC = () => {
+  const titleDefaultValue = '';
+  const userDefaultValue = 'Choose a user';
+
+  const [title, setTitle] = useState(titleDefaultValue);
+  const [user, setUser] = useState(userDefaultValue);
+  const [todos, setTodos] = useState(todoList);
+  const [isUserSelected, setIsUserSelected] = useState(true);
+  const [isTitleEntered, setIsTitleEntered] = useState(true);
+
+  const handleSubmitForm = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (user !== userDefaultValue && title !== titleDefaultValue) {
+      const newUser = getUserByName(user, usersFromServer);
```

Comment:
I think it is better (more convenient) to use selectedUserName here, because event.currentTarget.value from onChange event handler always returns string.